riscv64: adjust type definitions

CHAR8 needs to be defined while BOOLEAN should be removed
here to prevent typedef conflicts

Source: https://sourceforge.net/p/gnu-efi/code/ci/1de509497826faa0ad84b82f5e2c3d21ee613459/
---
--- a/inc/riscv64/efibind.h	2023-05-02 13:51:45.000000000 +0200
+++ -	2024-02-16 17:14:57.455180520 +0100
@@ -32,11 +32,9 @@
 typedef int16_t                 INT16;
 typedef uint8_t                 UINT8;
 typedef int8_t                  INT8;
+typedef char                    CHAR8;
 typedef wchar_t                 CHAR16;
 #define WCHAR                   CHAR16
-#ifndef BOOLEAN
-typedef uint8_t                 BOOLEAN;
-#endif
 #undef VOID
 typedef void                    VOID;
 typedef int64_t                 INTN;
